libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts

```diff
@@ -614,56 +614,96 @@ export class WorkspaceInstallManager extends BaseInstallManager {
+              installArgs.join(' ') +
+              '\n'
+          );
+        }
+
+        // Store the tip IDs that should be printed.
+        // They will be printed all at once *after* the install
+        const tipIDsToBePrinted: Set<CustomTipId> = new Set();
+        const pnpmTips: ICustomTipInfo[] = [];
+        for (const [customTipId, customTip] of Object.entries(PNPM_CUSTOM_TIPS)) {
+          if (
+            this.rushConfiguration.customTipsConfiguration.providedCustomTipsByTipId.has(
+              customTipId as CustomTipId
+            )
+          ) {
+            pnpmTips.push(customTip);
+          }
+        }
 
-            // Leave the pnpm-store as is for the retry. This ensures that packages that have already
-            // been downloaded need not be downloaded again, thereby potentially increasing the chances
-            // of a subsequent successful install.
+        const onPnpmStdoutChunk: ((chunk: string) => void) | undefined =
+          pnpmTips.length > 0
+            ? (chunk: string): void => {
+                // Iterate over the supported custom tip metadata and try to match the chunk.
+                for (const { isMatch, tipId } of pnpmTips) {
+                  if (isMatch?.(chunk)) {
+                    tipIDsToBePrinted.add(tipId);
+                  }
+                }
+              }
+            : undefined;
 
-            Utilities.createFolderWithRetry(splitWorkspaceNodeModulesFolder);
+        try {
+          await Utilities.executeCommandAndProcessOutputWithRetryAsync(
+            {
+              command: packageManagerFilename,
+              args: installArgs,
+              workingDirectory: this.rushConfiguration.commonTempSplitFolder,
+              environment: packageManagerEnv,
+              suppressOutput: false
+            },
+            this.options.maxInstallAttempts,
+            onPnpmStdoutChunk,
+            () => {
+              if (this.rushConfiguration.packageManager === 'pnpm') {
+                // eslint-disable-next-line no-console
+                console.log(colors.yellow(`Deleting the "node_modules" folder`));
+                this.installRecycler.moveFolder(splitWorkspaceNodeModulesFolder);
+
+                // Leave the pnpm-store as is for the retry. This ensures that packages that have already
+                // been downloaded need not be downloaded again, thereby potentially increasing the chances
+                // of a subsequent successful install.
+
+                Utilities.createFolderWithRetry(splitWorkspaceNodeModulesFolder);
+              }
+            }
+          );
+        } finally {
+          if (tipIDsToBePrinted.size > 0) {
+            this._terminal.writeLine();
+            for (const tipID of tipIDsToBePrinted) {
+              this.rushConfiguration.customTipsConfiguration._showTip(this._terminal, tipID);
+            }
           }
         }
-      );
+      };
+
+      await doInstalInternalSplitWorkspaceAsync();
```

Comment:
@theJiawen Interesting that `doInstalInternalSplitWorkspaceAsync()` was not called before this PR.  Was that a regression?